libp2p/protocols/pubsub/rpc/message.nim

```diff
@@ -77,9 +77,6 @@ proc init*(
     let peer = peer.get()
     msg.fromPeer = peer.peerId
     if sign:
-      if peer.keyType != KeyType.HasPrivate:
-        raise (ref LPError)(msg: "Cannot sign message without private key")
-
       msg.signature = sign(msg, peer.privateKey).expect("Couldn't sign message!")
       msg.key = peer.privateKey
```

Comment:
yaiks! this needs to be public key!

Reply:
Ah, `getKey` actually gives the publicKey, and since the chaining is on next line, it looks wrong here, but is actually giving the publicKey.
I'll do a PR to rename `getKey` to `getPublicKey` to avoid the confusion

Reply:
#621